Content.Shared/Whitelist/EntityWhitelistSystem.cs

```diff
@@ -23,6 +23,42 @@ public bool IsValid(EntityWhitelist list, [NotNullWhen(true)] EntityUid? uid)
         return uid != null && IsValid(list, uid.Value);
     }
 
+    /// <summary>
+    /// Checks whether a given entity satisfies a whitelist.
+    /// If the whitelist is null it returns null.
+    /// </summary>
+    public bool? Match(EntityWhitelist? list, [NotNullWhen(true)] EntityUid? uid)
+    {
+        if (list == null)
+            return null;
+
+        return IsValid(list, uid);
+    }
+
+    /// <summary>
+    /// Checks whether a given entity is allowed by a whitelist and not blocked by a blacklist.
+    /// If a blacklist is provided and it matches then this returns false.
+    /// If a whitelist is provided and it does not match then this returns false.
+    /// If either list is null it does not get checked.
+    /// </summary>
+    public bool IsAllowed([NotNullWhen(true)] EntityUid? uid, EntityWhitelist? blacklist = null, EntityWhitelist? whitelist = null)
+    {
+        if (uid == null)
+            return false;
+
+        return Match(blacklist, uid) != true
+            && Match(whitelist, uid) != false;
+    }
+
+    /// <summary>
+    /// Checks whether a given entity is on a blacklist.
+    /// If the blacklist is null it cannot be on it, so it returns false.
+    /// </summary>
+    public bool IsBlacklisted(EntityWhitelist? list, EntityUid? uid)
```

Comment:
This should probably just be called IsNotValid for congruency with the rest of the API

Reply:
forgor this was here meant to remove